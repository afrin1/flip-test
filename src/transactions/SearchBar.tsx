import React from "react"
import { StyleSheet, TextInput, View } from "react-native"
import { PrimaryButton } from "../components/Button"
import { colors } from "../theme"

interface SearchBarProps {
  onTextChanged: any,
  onButtonPressed: any,
  text: string
}

const SearchBar = (props: SearchBarProps) => {
  const { onTextChanged, onButtonPressed, text } = props

  return (
    <View style={styles.searchBar}>
      <TextInput
        placeholder="Cari nama, bank, atau nominal"
        placeholderTextColor={colors.GRAY}
        style={styles.search}
        onChangeText={onTextChanged}
        value={text}
        inlineImageLeft="search"
      />
      <PrimaryButton title={'URUTKAN V'} onPress={onButtonPressed} />
    </View>
  )
}

const styles = StyleSheet.create({
  search: {
    color: 'black',
    fontSize: 18,
    flex: 1,
  },
  searchBar: {
    marginTop: 10,
    marginLeft: 10,
    marginRight: 10,
    backgroundColor: '#FFFFFF',
    display: 'flex',
    flexDirection: 'row',
  }
})

export default SearchBar