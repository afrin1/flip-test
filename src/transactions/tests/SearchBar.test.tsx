import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import SearchBar from '../SearchBar'

describe('Search Bar', () => {

  const onTextChangedMock = jest.fn()
  const onButtonPressedMock = jest.fn()

  it('should match snapshot', () => {
    expect(render(<SearchBar 
      onTextChanged={onTextChangedMock}
      onButtonPressed={onButtonPressedMock}
      text={'Sample text'}
      />)).toMatchSnapshot()
  })

  it('should display the passed text', () => {
    const { getByTestId} = render(<SearchBar 
      onTextChanged={onTextChangedMock}
      onButtonPressed={onButtonPressedMock}
      text={'Sample Text'} />)

      expect(getByTestId('search-input')).toHaveProp('value', 'Sample Text')
  })

  it('should invoke the passed onTextChanged callback when input is typed', () => {
    const { getByTestId} = render(<SearchBar 
      onTextChanged={onTextChangedMock}
      onButtonPressed={onButtonPressedMock}
      text={'Sample Text'} />)

    fireEvent.changeText(getByTestId('search-input'), 'test')

    expect(onTextChangedMock).toBeCalled()
  })

  it('should invoke the onButtonPressed callback when Sort button is pressed', () => {
    const { getByTestId} = render(<SearchBar 
      onTextChanged={onTextChangedMock}
      onButtonPressed={onButtonPressedMock}
      text={'Sample Text'} />)

    fireEvent.press(getByTestId('sort-button'))

    expect(onButtonPressedMock).toBeCalled()
  })
})
