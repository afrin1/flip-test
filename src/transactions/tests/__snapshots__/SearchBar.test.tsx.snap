// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
    }
  }
>
  <TextInput
    inlineImageLeft="search"
    onChangeText={[MockFunction]}
    placeholder="Cari nama, bank, atau nominal"
    placeholderTextColor="#C5C5C5"
    style={
      Object {
        "color": "black",
        "flex": 1,
        "fontSize": 18,
      }
    }
    testID="search-input"
    value="Sample text"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="sort-button"
  >
    <Text
      style={
        Object {
          "color": "#FF6347",
          "fontSize": 15,
          "fontWeight": "bold",
          "margin": 10,
          "textAlignVertical": "center",
        }
      }
    >
      URUTKAN V
    </Text>
  </View>
</View>
`;
